### PR TITLE
docs: add new read-only states pattern

### LINKED ISSUE #3248: docs: read-only states pattern
## Read-only states pattern

Contributor docs box note: https://ibm.ent.box.com/notes/948323199814

----

### Content
Include these main sections with additional subsections:

- [x] Overview
- [x] When to use
- [x] When not to use
- [x] Formatting
  - [x] Anatomy
  - [x] Structure
  - [x] Best practices
  - [x] Interactions
- [x] Content
- [x] Accessibility
- [x] Related
- [x] References
- [x] Feedback

### Images
Revise any current images and add these additions:

- Add gallery images
- Add a do/don't image

### PR DESCRIPTION
**Pull request by laurenmrice**
Closes #3248 

Add a new pattern for read-only states with written guidance and images.

#### Changelog

**New**

- Content
- Images

### CODE REVIEW

**Comment by alisonjoseph** [comment]
Looks great! Do we want to index this for Platform?

**Review by tay1orjones — APPROVED** [review]

File: src/pages/patterns/read-only-states-pattern/index.mdx [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+
+<Caption>Read-only state anatomy.</Caption>
+
+</Column>
+</Row>
+
+1. **Background color change:** Having a transparent background for fields.
+2. **Border color change:** De-emphasizing selection and clickability to make
+   information more readable.
+3. **Icon color change:** Keepingg embedded icons in the component for context
```
Comment (aagonzales) [inline src/pages/patterns/read-only-states-pattern/index.mdx:81]:
```suggestion
3. **Icon color change:** Keeping embedded icons in the component for context
```

File: src/pages/patterns/read-only-states-pattern/index.mdx [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+   but displaying that icons are not interactive with color and cursor changes.
```
Comment (aagonzales) [inline src/pages/patterns/read-only-states-pattern/index.mdx:82]: I think we should add a 4th around the type
"Text color no change: text color remains the same as in the enabled state and should still pass 4.5:1 color contrast rules."

File: src/pages/patterns/read-only-states-pattern/index.mdx [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+## Visual guidance
+
+Read-only states use subtle changes to a component’s style to emphasize critical
+information and de-emphasize or remove icon signifiers.
+
+### Structure
+
+Components should maintain the same structure and spacing used in the
+component’s enabled state. In most cases, elements in the enabled state should
+be present in the read-only state.
```
Comment (aagonzales) [inline src/pages/patterns/read-only-states-pattern/index.mdx:93]:
```suggestion
component’s enabled state. In most cases, elements in the enabled state are present in the read-only state.
```

File: src/pages/patterns/read-only-states-pattern/index.mdx [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+
+#### Selection controls
+
+Selection controls offer users a selection from pre-determined options. Common
+selection controls include: checkboxes, radio buttons, toggles, dropdowns,
+selects, comboboxes and multiselects.
```
Comment (aagonzales) [inline src/pages/patterns/read-only-states-pattern/index.mdx:99]:
```suggestion
selects, combo boxes, and multiselects.
```

File: src/pages/patterns/read-only-states-pattern/index.mdx [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+<Caption>Read-only states for bound entry controls.</Caption>
+
+</Column>
+</Row>
+
+### Interactive elements
+
+Interactive elements included in a component’s enabled state should be removed
+or modified for its read-only state. Changes to color can be used to modify
+interactive elements and change their focus from interactive to informative.
```
Comment (aagonzales) [inline src/pages/patterns/read-only-states-pattern/index.mdx:131]: I would avoid the word "focus" outside the context of the focus state.

```suggestion
or modified for its read-only state. Changes to color are used to indicate a modification to
interactive elements and a change in their purpose from interactive to informative.
```

File: src/pages/patterns/read-only-states-pattern/index.mdx [hunk 6]
```diff
@@ -0,0 +1,360 @@
...
+
+</Tabs>
+
+</Column>
+</Row>
+
+#### Component icon signifiers
+
+Signifier icons included in a component’s enabled state, like chevron arrow
+icons, close icons, and calendar icons, should use the $icon-disabled color
```
Comment (aagonzales) [inline src/pages/patterns/read-only-states-pattern/index.mdx:195]:
```suggestion
icons, close icons, and calendar icons, should use the `$icon-disabled` color
```

File: src/pages/patterns/read-only-states-pattern/index.mdx [hunk 7]
```diff
@@ -0,0 +1,360 @@
...
+informative.
+
+<Row>
+<Column colLg={8}>
+
+<Tabs>
+
+<Tab label="Read-only">
+
+![Read-only content for a dropdown with no selection.](images/read-only-content.png)
```
Comment (aagonzales) [inline src/pages/patterns/read-only-states-pattern/index.mdx:298]: In this image, the field text color should still pass 4.5:1 contrast, change to text-secondary?
image

File: src/pages/patterns/read-only-states-pattern/index.mdx [hunk 8]
```diff
@@ -0,0 +1,360 @@
...
+</Column>
+</Row>
+
+## Accessibility
+
+When considering keyboard accessibility,  it can be helpful to distinguish
+between navigation and operation. A component that can be reached by keyboard is
+navigable. Read-only components remain navigable so that users can review the
+information they contain. (This contrasts with disabled components, which cannot
+be reached by keyboard.) However, read-only components are not operable—users
```
Comment (aagonzales) [inline src/pages/patterns/read-only-states-pattern/index.mdx:319]: That one sentence doesnt need the ( ) and I removed the em dash structure. 
```suggestion
information they contain. This contrasts with disabled components, which cannot be reached by a keyboard. However, read-only components are not operable, meaning users
```

**Review by aagonzales — CHANGES_REQUESTED** [review]
This looks awesome, you did a great job bringing this across the finish line. I just had a few content suggestions to help with clarity in places.

**Review by alisonjoseph — APPROVED** [review]

**Review by alisonjoseph — APPROVED** [review]

File: src/pages/patterns/read-only-states-pattern/index.mdx [hunk 9]
```diff
@@ -0,0 +1,360 @@
...
+title: Read-only states
+description:
+  Read-only states are applied to components that users can review but not
+  modify and remove a component’s interactive functions.
+---
+
+<PageDescription>
+
+Read-only states are applied to components that users can review but not modify
+and remove a component’s interactive functions.
```
[the remark this answers is not in the document]
Comment (laurenmrice) [reply src/pages/patterns/read-only-states-pattern/index.mdx]: Agreed! Thank you.

File: src/pages/patterns/read-only-states-pattern/index.mdx (hunk 2 above)
Comment (laurenmrice) [reply src/pages/patterns/read-only-states-pattern/index.mdx:78]: Good idea! Will add 👍

File: src/pages/patterns/read-only-states-pattern/index.mdx (hunk 7 above)
Comment (laurenmrice) [reply src/pages/patterns/read-only-states-pattern/index.mdx:293]: Ok yes, I thought something was off here. I will try out `$text-secondary`.

**Comment by laurenmrice** [comment]
@aagonzales Okay, all feedback has been added! Thank you.

— I update the read-only state colors with the latest spec for all images.
— I updated the do/don't image with a new UI.
— I also added the fluid variant in images when I am using the tab component for comparisons between the read-only/enabled states since they have some differences. Let me know what you think about that.

File: src/pages/patterns/read-only-states-pattern/index.mdx [hunk 10]
```diff
@@ -0,0 +1,365 @@
...
+title: Read-only states
+description:
+  Read-only states are applied to components that users can review but not
+  modify and remove a component’s interactive functions.
+---
+
+<PageDescription>
+
+Read-only states are applied to components when the user is allowed to review
+but not modify the component. It removed all interactive functions from the
```
Comment (aagonzales) [inline src/pages/patterns/read-only-states-pattern/index.mdx:11]: oops this is my own typo, should this be "removes" instead?
```suggestion
but not modify the component. It removes all interactive functions from the
```

Comment (laurenmrice) [reply src/pages/patterns/read-only-states-pattern/index.mdx:11]: Oh, yep! Good catch.

**Comment by laurenmrice** [comment]
— Added a notification for fluid components.

**Review by aagonzales — APPROVED** [review]
Ship it!! Looks good!

**Pull request merged**